src/main/java/com/twilio/oai/resolver/LanguageParamResolver.java

```diff
@@ -0,0 +1,74 @@
+import com.twilio.oai.resolver.php.IConventionMapper;
+import com.twilio.oai.resolver.php.LanguageConventionResolver;
+import org.openapitools.codegen.CodegenParameter;
+
+public class LanguageParamResolver implements IResolver<CodegenParameter> {
+    protected IConventionMapper mapper;
+    public final static String SERIALIZE_VEND_EXT = "x-serialize";
+    public final static String DESERIALIZE_VEND_EXT = "x-deserialize";
+
+    public LanguageParamResolver(IConventionMapper mapper) {
+        this.mapper = mapper;
+    }
+
+    @Override
+    public CodegenParameter resolve(CodegenParameter codegenParameter) {
+        resolveProperties(codegenParameter);
+        resolvePrefixedMap(codegenParameter);
+        resolveSerialize(codegenParameter);
+        resolveDeSerialize(codegenParameter);
+        return codegenParameter;
+    }
+
+    @Override
+    public void resolveProperties(CodegenParameter codegenParameter) {
+        if (mapper.properties().containsKey(codegenParameter.dataFormat)) {
+            codegenParameter.dataType = (String) mapper.properties().get(codegenParameter.dataFormat);
+        }
+    }
+
+    @Override
+    public void resolveSerialize(CodegenParameter codegenParameter) {
+        boolean hasProperty = mapper.serialize().containsKey(codegenParameter.dataFormat);
+        if (hasProperty) {
+            codegenParameter.vendorExtensions.put(SERIALIZE_VEND_EXT, (String) mapper.serialize().get(codegenParameter.dataFormat));
+            codegenParameter.vendorExtensions.put(SERIALIZE_VEND_EXT + LanguageConventionResolver.HYPHEN + codegenParameter.dataFormat, true);
+        }
+        boolean hasType = mapper.serialize().containsKey(codegenParameter.dataType);
+        if (hasType) {
+            codegenParameter.vendorExtensions.put(SERIALIZE_VEND_EXT, (String) mapper.serialize().get(codegenParameter.dataType));
+            codegenParameter.vendorExtensions.put(SERIALIZE_VEND_EXT + LanguageConventionResolver.HYPHEN + codegenParameter.dataType, true);
+        }
+    }
+
+    @Override
+    public void resolveDeSerialize(CodegenParameter codegenParameter) {
+        if (mapper.deserialize().containsKey(codegenParameter.dataFormat)) {
+            codegenParameter.vendorExtensions.put(DESERIALIZE_VEND_EXT, (String) mapper.deserialize().get(codegenParameter.dataFormat));
+            codegenParameter.vendorExtensions.put(DESERIALIZE_VEND_EXT + LanguageConventionResolver.HYPHEN + codegenParameter.dataType, true);
+        }
+
+        if (mapper.deserialize().containsKey(codegenParameter.dataType)) {
+            codegenParameter.vendorExtensions.put(DESERIALIZE_VEND_EXT, (String) mapper.serialize().get(codegenParameter.dataType));
+            codegenParameter.vendorExtensions.put(DESERIALIZE_VEND_EXT + LanguageConventionResolver.HYPHEN + codegenParameter.dataType, true);
+        }
+    }
+
+
+
+    @Override
+    public void resolvePrefixedMap(CodegenParameter codegenParameter) {
```

Comment:
Do all these need to be in the interface and public?

Reply:
I thought this through and decided to add 4/5 resolve methods to Language<param/property>Resolver, It will allow each helper to extend basic functionality and on top of that to add complexType and enum logic. I get what you are coming from, I'll change it to protected and keep it part of the Language abstract class and trim interface to resolve method.